File: archive/cold-storage/reference/aaes_os_v1/typescript/src/policy_engine.ts
import type { AAESContext, AAESDecision, AAESRequest, PolicyEngine } from "./types.js";
import { notImplemented } from "./error.js";

export class PolicyEngineStub implements PolicyEngine {
  evaluate(_request: AAESRequest, _context: AAESContext) {
    return notImplemented<AAESDecision>("PolicyEngine.evaluate");
  }
}

File: archive/cold-storage/reference/aaes_os_v1/rust/src/policy_engine.rs
use crate::error::{not_implemented, AaesResult};
use crate::orchestrator::AAESContext;
use crate::types::{AAESDecision, AAESRequest};

#[derive(Debug, Default)]
pub struct PolicyEngineStub;

impl PolicyEngineStub {
    pub fn evaluate(
        &self,
        _request: &AAESRequest,
        _context: &AAESContext,
    ) -> AaesResult<AAESDecision> {
        Err(not_implemented("PolicyEngine.evaluate"))
    }
}
